--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -799,6 +799,13 @@ static int hostapd_ctrl_iface_get_config
 		pos += ret;
 	}
 
+    ret = os_snprintf(pos, end - pos, "ignore_broadcast_ssid=%d\n",
+        hapd->conf->ignore_broadcast_ssid);
+
+    if (ret < 0 || ret >= end - pos)
+        return pos - buf;
+    pos += ret;
+
 	return pos - buf;
 }
 
@@ -864,7 +871,12 @@ static int hostapd_ctrl_iface_get(struct
 		if (res < 0 || (unsigned int) res >= buflen)
 			return -1;
 		return res;
-	}
+	} else if (os_strcmp(cmd, "ignore_broadcast_ssid") == 0) {
+        res = os_snprintf(buf, buflen, "%d", hapd->conf->ignore_broadcast_ssid);
+        if (res < 0 || (unsigned int) res >= buflen)
+            return -1;
+        return res;
+    }
 
 	return -1;
 }
