--- a/hostapd/ctrl_iface.c
+++ b/hostapd/ctrl_iface.c
@@ -29,6 +29,7 @@
 #include "ap/wps_hostapd.h"
 #include "ap/ctrl_iface_ap.h"
 #include "ap/ap_drv_ops.h"
+#include "ap/beacon.h"
 #include "wps/wps_defs.h"
 #include "wps/wps.h"
 #include "config_file.h"
@@ -853,6 +854,10 @@ static int hostapd_ctrl_iface_set(struct
 #endif /* CONFIG_INTERWORKING */
 	} else {
 		ret = hostapd_set_iface(hapd->iconf, hapd->conf, cmd, value);
+        if (os_strcasecmp(cmd, "ignore_broadcast_ssid") == 0) {
+            ieee802_11_set_beacon(hapd);
+        }
+
 	}
 
 	return ret;
